src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -162,11 +195,38 @@ impl WfContext {
 
     /// Request to run a local activity
     pub fn local_activity(
-        &mut self,
+        &self,
+        opts: LocalActivityOptions,
+    ) -> impl CancellableFuture<ActivityResolution> + '_ {
+        // We need to immediately create the first LA future to be consistent with the behavior
+        // of other functions where the command is scheduled before being awaited on.
+        let first_la_fut = self.local_activity_no_timer_retry(opts.clone());
+        ManualCancellableFuture::new(|_ct| async move {
+            let mut last_resolution = first_la_fut.await;
+
+            while let Some(activity_resolution::Status::Backoff(b)) = last_resolution.status {
+                self.timer(
+                    b.backoff_duration
+                        .expect("Duration is set")
+                        .try_into()
+                        .expect("duration converts ok"),
+                )
+                .await;
+                let mut opts = opts.clone();
+                opts.attempt = Some(b.attempt);
+                last_resolution = self.local_activity_no_timer_retry(opts.clone()).await;
+            }
+            last_resolution
+        })
+    }
+
+    /// Request to run a local activity with no implementation of timer-backoff based retrying.
+    /// TODO: Should not be public in non-prototype SDK
+    pub(crate) fn local_activity_no_timer_retry(
```

Comment:
This could probably just be private, I don't see a reason to use it anywhere.
Even the test that used it should probably just use `local_activity` directly.

Reply:
The unit test is able to verify directly that the right response is returned for telling lang to back off, but I can probably do that lower level in the machine test so I can change this one.